travis/gen_addons_table.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/env/bin/python
+from __future__ import print_function
+import ast
+import os
+import re
+
+
+MARKERS = r'(\[//\]: # \(addons\))|(\[//\]: # \(end addons\))'
+
+
+class UserError(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+
+def sanitize_cell(s):
+    if not s:
+        return ''
+    s = ' '.join(s.split())
+    return s
+
+
+def render_markdown_table(header, rows):
+    table = []
+    rows = [header, ['---'] * len(header)] + rows
+    for row in rows:
+        table.append(' | '.join(row))
+    return '\n'.join(table)
+
+
+def replace_in_readme(readme_path, header, rows_available, rows_unported):
+    readme = open(readme_path).read()
+    parts = re.split(MARKERS, readme, flags=re.MULTILINE)
+    if len(parts) != 7:
+        raise UserError('Addons markers not found or incorrect in %s' %
+                        readme_path)
+    addons = []
+    if rows_available:
+        addons.extend([
+            '\n',
+            'Available addons\n',
+            '----------------\n',
+            render_markdown_table(header, rows_available),
+            '\n'
+        ])
+    if rows_unported:
+        addons.extend([
+            '\n',
+            'Unported addons\n',
+            '---------------\n',
+            render_markdown_table(header, rows_unported),
+            '\n'
+        ])
+    parts[2:5] = addons
+    readme = ''.join(parts)
+    open(readme_path, 'w').write(readme)
+
+
+def gen_addons_table():
+    readme_path = 'README.md'
```

Comment:
@pedrobaeza the script preserves the placeholder in the newly generated README, so you can run it several times.